protected/controllers/BaseEventTypeController.php

```diff
@@ -274,14 +274,26 @@ public function renderEventMetadata($view = '//patient/event_metadata')
     public function getElements()
     {
         $elements = array();
+        $parent_element_ids = array();
         if (is_array($this->open_elements)) {
             foreach ($this->open_elements as $element) {
                 if ($element->getElementType() && !$element->getElementType()->isChild()) {
                     $elements[] = $element;
                 }
+                elseif ($element->getElementType() && $element->getElementType()->isChild()){
+                    if (!in_array($element->getElementType()->parent_element_type_id, $parent_element_ids)){
+                        $parent_element_ids[] = $element->getElementType()->parent_element_type_id;
+                    }
+                }
+            }
+            foreach ($parent_element_ids as $key=>$p_id){
+                $parent_ele = ElementType::model()->findAllByPk($p_id)[0];
```

Comment:
If you're only after a single ElementType, wouldn't it be easier to call the findByPk function and return a single element rather than get the first element from findAllByPk?

Reply:
Yeah, I didn't notice there are two different functions.